### PR TITLE
Update CI

### PR DESCRIPTION
**Pull request by ptoffy**
- Show warnings as errors
- Add benchmark CI using https://github.com/ordo-one/package-benchmark

### CODE REVIEW

**Review by 0xTim — APPROVED** [review]

**Comment by 0xTim** [comment]
@ptoffy this will need rebasing

File: .github/workflows/benchmark.yml [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+      - name: Comment PR
+        if: ${{ env.hasBenchmark == '1' }}
+        uses: thollander/actions-comment-pull-request@v2
+        with:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          message: ${{ env.PRTEST }}
+          comment_includes: "Pull request benchmark comparison [${{ matrix.os }}] with"
+      - name: Exit with correct status
+        run: |
+          exit ${{ env.exitStatus }}
```
Comment (gwynne) [inline .github/workflows/benchmark.yml:89]:
```suggestion
name: Benchmark PR vs main
on:
  workflow_dispatch:
  pull_request:
    branches: [ main ]
  
jobs:
  benchmark-delta-linux:
    runs-on: ubuntu-latest
    container: swift:jammy
    steps:
      - uses: actions/checkout@v4
      - name: jemalloc dependency
        run: sudo apt-get update && sudo apt-get install -y libjemalloc-dev
      - name: Run benchmarks for PR branch
        if: ${{ hashFiles('Benchmarks/**') != '' }}
        run: |
          swift package --allow-writing-to-directory .benchmarkBaselines/ benchmark baseline update pull_request
          git checkout main
          swift package --allow-writing-to-directory .benchmarkBaselines/ benchmark baseline update main
          date >>"${GITHUB_STEP_SUMMARY}"
          swift package benchmark baseline check main pull_request --format markdown >>"${GITHUB_STEP_SUMMARY}"
        uses: thollander/actions-comment-pull-request@v2
        with:
          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
          message: ${{ format('[PR benchmark comparison Linux|main]({0}/{1}/actions/runs/{2})', github.server_url, github.repository, github.run_id) }}
          comment_tag: 'PR benchmark comparison Linux'
```

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -25,7 +25,7 @@ jobs:
       - name: Use local JWTKit
         run: swift package --package-path jwt edit jwt-kit --path ./jwt-kit
       - name: Run tests with Thread Sanitizer
-        run: swift test --package-path jwt --sanitize=thread
+        run: swift test --package-path jwt --sanitize=thread -Xswiftc -warnings-as-errors
```
Comment (gwynne) [inline .github/workflows/test.yml:28]: This is a good way to make sure CI never ever passes, given the way warnings tend to show up without warning (no pun intended) thanks to changes in dependencies.

**Review by gwynne — CHANGES_REQUESTED** [review]

**Comment by ptoffy** [comment]
Closed in favour of #141

**Pull request closed without merging**